TO BE MERGED FOR 2.8.2 - MTV-2332: Update MTV recommendations for migrating VMS with large disks (#671)
kubev2v/forklift-documentation: documentation/modules/mtv-performance-recommendation.adoc

* *Maximum transfer rate observed of multiple disks from a single ESXi host: 253 MB/s (concurrent migration of 10 VMs, 1 disk each, 35/50 GiB used per disk)*
* *Total transfer rate observed of multiple disks (210) from 8 ESXi hosts: 802 MB/s (concurrent migration of 70 VMs, 3 disks each, 35/50 GiB used per disk)*

[id="mtv-precommendations-vms-large-disks_{context}"]
== Recommendations for migrating VMs with large disks

The following recommendations are suggested for VMs with data on disk totaling to 1 TB or greater for each individual disk:

> 

Reviewer: Lets add this as the first point:

“Schedule appropriate maintenance windows for migrating large disk VMs. Those migrations are sensitive operations and may require careful planning of the maintenance windows and downtime, ideally during periods of lower storage and network activity.”

Reviewer 2: +1

> 

Author: ```suggestion

* Schedule appropriate maintenance windows for migrating large disk virtual machines (VMs). Such migrations are sensitive operations and may require careful planning of maintenance windows and downtime, especially during periods of lower storage and network activity.
```

> * Make sure that no other migration activities or other heavy network or storage activities run during those large virtual machine (VM) migrations, and treat them as a special case. During those migrations, prioritize {project-short} activities. You should plan to migrate those VMs to a time when there are fewer activities on those VMs and related datastore.

Author: ```suggestion
* Check that no other migration activities or other heavy network or storage activities are run during those large virtual machine (VM) migrations. You should treat these large virtual machine migrations as a special case. During those migrations, prioritize {project-short} activities. You should plan to migrate those VMs to a time when there are fewer activities on those VMs and related datastore.
```

> * Reduce the warm migration `controller_precopy_interval` to 15 minutes; the default is 60 minutes. Schedule the cut-over to be done at least 24 hours before the maintenance window. Schedule the cut-over to the beginning of that maintenance window. This schedule should allow the last snapshots to be small enough, including only the changes from the last 15 minutes.

Reviewer: For Large VMs with a high churn rate, meaning data is frequently changed in amounts of 100GB or more between snapshots, consider reducing the warm migration controller_precopy_interval from the default which is 60 minutes.  
 
Ensure that this process is started at least 24 hours before the scheduled cut-over in order to allow for multiple successful precopy snapshots to complete.  When scheduling the cut-over, ensure that the maintenance window will allow for enough time for the last snapshot of changes to be copied over and that the cut-over process begins at the beginning of that maintenance window.

Reviewer 2: +1

> * Reduce the warm migration `controller_precopy_interval` to 15 minutes; the default is 60 minutes. Schedule the cut-over to be done at least 24 hours before the 3-hour maintenance window. Schedule the cut-over to the beginning of that maintenance window. This schedule should allow the last snapshots to be small enough, including only the changes from the last 15 minutes.

Reviewer 2: Remove 3h - (Each customer has different mintance windows)

> * Reduce the warm migration `controller_precopy_interval` to 15 minutes; the default is 60 minutes. Schedule the cut-over to be done at least 24 hours before the 3-hour maintenance window. Schedule the cut-over to the beginning of that maintenance window. This schedule should allow the last snapshots to be small enough, including only the changes from the last 15 minutes.

Author: ```suggestion
* Reduce the warm migration `controller_precopy_interval` to 15 minutes; the default is 60 minutes. Schedule the cut-over to be done at least 24 hours before the maintenance window. Schedule the cut-over to the beginning of that maintenance window. This schedule should allow the last snapshots to be small enough, including only the changes from the last 15 minutes.
```

* In cases of particularly large single-disk VMs, where some downtime is possible, select cold migrations rather than warm migrations, especially in the case of large VM snapshots.

* Consider splitting data on particularly large disks to multiple disks, which enables parallel disk migration with {project-short} when warm migration is used.

* If you have large database disks with continuous writes of large amounts of data, where downtime and VM snapshots are not possible, it may be necessary to consider database vendor-specific replication options of the database data to target these specific migrations outside {project-short}. Please consult the vendor-specific options of your database if this case applies.

> * Plan proper maintenance windows for large disk VM migrations, as they are sensitive operations, and scheduling migrations for downtime windows during less storage-intensive operations is essential.

Reviewer: This statement can be removed

" Plan proper maintenance windows for large disk VM migrations, as they are sensitive operations, and scheduling migrations for downtime windows during less storage-intensive operations is essential." 

its replaced by the first comment in this review where the statement was reworded and moved up in the list.

* *Maximum transfer rate observed of multiple disks from a single ESXi host: 253 MB/s (concurrent migration of 10 VMs, 1 disk each, 35/50 GiB used per disk)*
* *Total transfer rate observed of multiple disks (210) from 8 ESXi hosts: 802 MB/s (concurrent migration of 70 VMs, 3 disks each, 35/50 GiB used per disk)*

[id="mtv-precommendations-vms-large-disks_{context}"]
== Recommendations for migrating VMs with large disks

The following recommendations are suggested for VMs with data on disk totaling to 1 TB or greater for each individual disk:

> Schedule appropriate maintenance windows for migrating large disk virtual machines (VMs). Such migrations are sensitive operations and may require careful planning of maintenance windows and downtime, especially during periods of lower storage and network activity.

Author: ```suggestion
* Schedule appropriate maintenance windows for migrating large disk virtual machines (VMs). Such migrations are sensitive operations and may require careful planning of maintenance windows and downtime, especially during periods of lower storage and network activity.
```

> * Check that no other migration activities or other heavy network or storage activities are run during those large virtual machine (VM) migrations. You should treat these large virtual machine migrations as a special case. During those migrations, prioritize {project-short} activities. Plan to migrate those VMs to a time when there are fewer activities on those VMs and related datastore.

Author: ```suggestion

* Check that no other migration activities or other heavy network or storage activities are run during those large virtual machine (VM) migrations. You should treat these large virtual machine migrations as a special case. During those migrations, prioritize {project-short} activities. Plan to migrate those VMs to a time when there are fewer activities on those VMs and related datastore.
```

* Check that no other migration activities or other heavy network or storage activities are run during those large virtual machine (VM) migrations. You should treat these large virtual machine migrations as a special case. During those migrations, prioritize {project-short} activities. Plan to migrate those VMs to a time when there are fewer activities on those VMs and related datastore.

> * For large VMs with a high churn rate, which means data is frequently changed in amounts of 100 GB or more between snapshots, consider reducing the warm migration `controller_precopy_interval` from the default, which is 60 minutes. It is important to ensure that this process is started at least 24 hours before the scheduled cutover to allow for multiple successful precopy snapshots to complete. When scheduling the cut-over, ensure that the maintenance window allows for sufficient time for the last snapshot of changes to be copied over and that the cutover process begins at the beginning of that maintenance window.

Author: ```suggestion
* For large VMs with a high churn rate, which means data is frequently changed in amounts of 100 GB or more between snapshots, consider reducing the warm migration `controller_precopy_interval` from the default, which is 60 minutes. It is important to ensure that this process is started at least 24 hours before the scheduled cutover to allow for multiple successful precopy snapshots to complete. When scheduling the cutover, ensure that the maintenance window allows for sufficient time for the last snapshot of changes to be copied over and that the cutover process begins at the beginning of that maintenance window.
```

> * For large VMs with a high churn rate, which means data is frequently changed in amounts of 100 GB or more between snapshots, consider reducing the warm migration `controller_precopy_interval` from the default, which is 60 minutes. It is important to ensure that this process is started at least 24 hours before the scheduled cut-over to allow for multiple successful precopy snapshots to complete. When scheduling the cut-over, ensure that the maintenance window allows for sufficient time for the last snapshot of changes to be copied over and that the cut-over process begins at the beginning of that maintenance window.

Author: ```suggestion
* For large VMs with a high churn rate, which means data is frequently changed in amounts of 100 GB or more between snapshots, consider reducing the warm migration `controller_precopy_interval` from the default, which is 60 minutes. It is important to ensure that this process is started at least 24 hours before the scheduled cutover to allow for multiple successful precopy snapshots to complete. When scheduling the cut-over, ensure that the maintenance window allows for sufficient time for the last snapshot of changes to be copied over and that the cutover process begins at the beginning of that maintenance window.
```

* *Maximum transfer rate observed of multiple disks from a single ESXi host: 253 MB/s (concurrent migration of 10 VMs, 1 disk each, 35/50 GiB used per disk)*
* *Total transfer rate observed of multiple disks (210) from 8 ESXi hosts: 802 MB/s (concurrent migration of 70 VMs, 3 disks each, 35/50 GiB used per disk)*

[id="mtv-precommendations-vms-large-disks_{context}"]
== Recommendations for migrating VMs with large disks

The following recommendations are suggested for VMs with data on disk totaling to 1 TB or greater for each individual disk:

> * Schedule appropriate maintenance windows for migrating large disk virtual machines (VMs). Such migrations are sensitive operations and may require careful planning of maintenance windows and downtime, especially during periods of lower storage and network activity.

Author: ```suggestion
Schedule appropriate maintenance windows for migrating large disk virtual machines (VMs). Such migrations are sensitive operations and may require careful planning of maintenance windows and downtime, especially during periods of lower storage and network activity.
```

* Reduce the warm migration `controller_precopy_interval` to 15 minutes; the default is 60 minutes. Schedule the cut-over to be done at least 24 hours before the maintenance window. Schedule the cut-over to the beginning of that maintenance window. This schedule should allow the last snapshots to be small enough, including only the changes from the last 15 minutes.

* In cases of particularly large single-disk VMs, where some downtime is possible, select cold migrations rather than warm migrations, especially in the case of large VM snapshots.

* Consider splitting data on particularly large disks to multiple disks, which enables parallel disk migration with {project-short} when warm migration is used.

* If you have large database disks with continuous writes of large amounts of data, where downtime and VM snapshots are not possible, it may be necessary to consider database vendor-specific replication options of the database data to target these specific migrations outside {project-short}. Please consult the vendor-specific options of your database if this case applies.

> * Plan proper maintenance windows for large disk VM migrations, as they are sensitive operations, and scheduling migrations for downtime windows during less storage-intensive operations is essential.

Author: ```suggestion
```

* Check that no other migration activities or other heavy network or storage activities are run during those large virtual machine (VM) migrations. You should treat these large virtual machine migrations as a special case. During those migrations, prioritize {project-short} activities. Plan to migrate those VMs to a time when there are fewer activities on those VMs and related datastore.

* For large VMs with a high churn rate, which means data is frequently changed in amounts of 100 GB or more between snapshots, consider reducing the warm migration `controller_precopy_interval` from the default, which is 60 minutes. It is important to ensure that this process is started at least 24 hours before the scheduled cut-over to allow for multiple successful precopy snapshots to complete. When scheduling the cut-over, ensure that the maintenance window allows for sufficient time for the last snapshot of changes to be copied over and that the cut-over process begins at the beginning of that maintenance window.

> +
> This schedule should allow the last snapshots to be small enough, including only the changes from the last 15 minutes.

Author: ```suggestion
```

> +
> This schedule should allow the last snapshots to be small enough, including only the changes from the last 15 minutes.

Author: ```suggestion
```

* For large VMs with a high churn rate, which means data is frequently changed in amounts of 100 GB or more between snapshots, consider reducing the warm migration `controller_precopy_interval` from the default, which is 60 minutes. It is important to ensure that this process is started at least 24 hours before the scheduled cut-over to allow for multiple successful precopy snapshots to complete. When scheduling the cut-over, ensure that the maintenance window allows for sufficient time for the last snapshot of changes to be copied over and that the cut-over process begins at the beginning of that maintenance window.

** Reduce the warm migration `controller_precopy_interval` to 15 minutes. 
** Schedule the cut-over to be done at least 24 hours before the maintenance window. 
** Schedule the cut-over to the beginning of that maintenance window. 

> This schedule should allow the last snapshots to be small enough, including only the changes from the last 15 minutes.

Author: ```suggestion
+
This schedule should allow the last snapshots to be small enough, including only the changes from the last 15 minutes.
```

* Check that no other migration activities or other heavy network or storage activities are run during those large virtual machine (VM) migrations. You should treat these large virtual machine migrations as a special case. During those migrations, prioritize {project-short} activities. Plan to migrate those VMs to a time when there are fewer activities on those VMs and related datastore.

* For large VMs with a high churn rate, which means data is frequently changed in amounts of 100 GB or more between snapshots, consider reducing the warm migration `controller_precopy_interval` from the default, which is 60 minutes. It is important to ensure that this process is started at least 24 hours before the scheduled cut-over to allow for multiple successful precopy snapshots to complete. When scheduling the cut-over, ensure that the maintenance window allows for sufficient time for the last snapshot of changes to be copied over and that the cut-over process begins at the beginning of that maintenance window.

> 
> ** Reduce the warm migration `controller_precopy_interval` to 15 minutes. 
> ** Schedule the cut-over to be done at least 24 hours before the maintenance window. 
> ** Schedule the cut-over to the beginning of that maintenance window. 
> 

Author: ```suggestion
```

+

> This schedule should allow the last snapshots to be small enough, including only the changes from the last 15 minutes.

Reviewer 2: Remove  those 4 rows  - (as it is a repetition of what is written i the paragraph above and I don[t want to recomend 15 minuts as for some customers we saw 15 munuts is to low, They will need to find the golden interval  that fits them. We just recommend to reduce)

"
**Reduce the warm migration `controller_precopy_interval` to 15 minutes. 
** Schedule the cut-over to be done at least 24 hours before the maintenance window. 
** Schedule the cut-over to the beginning of that maintenance window. 
This schedule should allow the last snapshots to be small enough, including only the changes from the last 15 minutes.
+
